packages/signature-controller/src/SignatureController.test.ts

```diff
@@ -416,7 +430,41 @@ describe('SignatureController', () => {
       });
 
       it('approves message and signs', async () => {
-        await (signatureController as any)[signMethodName](messageParamsMock);
+        await new Promise(async (resolve) => {
```

Comment:
Rather than wrapping the test in a promise to bind it to the event, would it be simpler to use a `jest.fn()` for the event listener?

That way the test can simply assert as normal at the end using an `expect(listenerMock).toHaveBeenCalledTimes(1)` for example?

Reply:
Much more readable that way - I've used the pattern you suggested in the new test